features/promotion/receiving_discount/receiving_discounts_with_product_minimum_price_specified.feature

```diff
@@ -0,0 +1,104 @@
+        When I add product "T-Shirt" to the cart
+        Then its price should be decreased by "$5.00"
+        And my cart total should be "$45.00"
+
+    @ui @api
+    Scenario: Distributing fixed discount promotion
+        And this promotion gives "$10.00" off on every product with minimum price at "$10.00"
+        When I add product "T-Shirt" to the cart
+        And I add product "PHP Mug" to the cart
+        Then product "T-Shirt" price should be decreased by "$5.00"
+        And product "PHP Mug" price should be decreased by "$10.00"
+        And my cart total should be "$55.00"
+
+    @ui @api
+    Scenario: Receiving fixed discount for my cart
+        And the promotion gives "$10.00" discount to every order with items total at least "$20.00"
+        When I add product "T-Shirt" to the cart
+        Then my cart total should be "$45.00"
+        And my discount should be "-$5.00"
+
+    @ui @api
+    Scenario: Receiving discount when buying more than required quantity
+        Given there is a promotion "T-Shirts promotion"
+        And the promotion gives "$50.00" discount to every order with quantity at least 2
+        When I add 2 products "T-Shirt" to the cart
+        Then my cart total should be "$90.00"
+        And my discount should be "-$10.00"
+
+    @api
+    Scenario: Distributing fixed order discount promotion
+        Given the promotion gives "$20.00" discount to every order with quantity at least 2
+        When I add product "T-Shirt" to the cart
+        And I add product "PHP Mug" to the cart
+        Then product "T-Shirt" price should be decreased by "$5.00"
+        And product "PHP Mug" price should be decreased by "$15.00"
+        And my cart total should be "$50.00"
+
+    @api
+    Scenario: Distributing percentage order discount promotion
+        Given it gives "20%" discount to every order
+        When I add product "T-Shirt" to the cart
+        And I add product "PHP Mug" to the cart
+        Then product "T-Shirt" price should be decreased by "$5.00"
+        And product "PHP Mug" price should be decreased by "$9.00"
+        And my cart total should be "$56.00"
+
+    @api
+    Scenario: Distributing discount evenly between different products when one has minimum price specified
+        Given it gives "$25" discount to every order
+        And I add product "T-Shirt" to the cart
+        And I add 3 products "PHP Mug" to the cart
+        And I specified the billing address as "Ankh Morpork", "Frost Alley", "90210", "United States" for "Jon Snow"
+        When I proceed with "Free" shipping method and "Offline" payment
+        Then I should be on the checkout summary step
+        And the "T-Shirt" product should have unit price discounted by "$5.00"
+        And the "PHP Mug" product should have unit prices discounted by "$6.67", "$6.67" and "$6.66"
+
+    @api
+    Scenario: Distributing discount evenly between different products when one has minimum price specified
```

Comment:
```suggestion
    Scenario: Distributing discount proportionally between different products when one has minimum price specified
```
it would be a better word than `evenly`, and it got me thinking that we should add one more scenario to check proportionally distributions by adding one more product without a minimum price and with a different price than PHP Mug

Reply:
I suggest adding a scenario:
```
    Background:
        ...
        And the store has a product "Symfony Mug" priced at "$40.00"

    @api
    Scenario: Distributing discount proportionally between different products when one has minimum price specified
        Given it gives "$27" discount to every order
        And I add 2 products "T-Shirt" to the cart
        And I add 2 products "PHP Mug" to the cart
        And I add product "Symfony Mug" to the cart
        And I specified the billing address as "Ankh Morpork", "Frost Alley", "90210", "United States" for "Jon Snow"
        When I proceed with "Free" shipping method and "Offline" payment
        Then I should be on the checkout summary step
        And the "T-Shirt" product should have unit prices discounted by "$5.00" and "$5.00"
        And the "PHP Mug" product should have unit prices discounted by "$4.25" and "$4.25"
        And the "Symfony Mug" product should have unit price discounted by "$8.5"
```
and it doesn't work right now I guess